src/Keboola/Console/Command/MassProjectQueueMigration.php

```diff
@@ -0,0 +1,187 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\Console\Command;
+
+
+use Keboola\JobQueueClient\Client as JobQueueClient;
+use Keboola\JobQueueClient\JobData;
+use Keboola\ManageApi\Client;
+use Keboola\StorageApi\Client as StorageClient;
+use Keboola\StorageApi\Components;
+use Keboola\StorageApi\Options\Components\Configuration;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Logger\ConsoleLogger;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class MassProjectQueueMigration extends Command
+{
+    const ARGUMENT_MANAGE_TOKEN = 'manage-token';
+    const ARGUMENT_CONNECTION_URL = 'connection-url';
+    const ARGUMENT_SOURCE_FILE = 'source-file';
+
+    const FEATURE_QUEUE_V2 = 'queuev2';
+    const COMPONENT_QUEUE_MIGRATION_TOOL = 'keboola.queue-migration-tool';
+    const JOB_STATES_FINAL = ['success', 'error', 'terminated', 'cancelled'];
+
+    protected function configure()
+    {
+        $this
+            ->setName('manage:mass-project-queue-migration')
+            ->setDescription('Mass project migration to Queue v2')
+            ->addArgument(self::ARGUMENT_MANAGE_TOKEN, InputArgument::REQUIRED, 'Manage token')
+            ->addArgument(self::ARGUMENT_CONNECTION_URL, InputArgument::REQUIRED, 'Connection url')
+            ->addArgument(self::ARGUMENT_SOURCE_FILE, InputArgument::REQUIRED, 'Source file with project ids')
+        ;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $manageToken = $input->getArgument(self::ARGUMENT_MANAGE_TOKEN);
+        $kbcUrl = $input->getArgument(self::ARGUMENT_CONNECTION_URL);
+        $sourceFile = $input->getArgument(self::ARGUMENT_SOURCE_FILE);
+        $output->writeln(sprintf('Fetching projects from "%s"', $sourceFile));
+
+        $manageClient = new Client([
+            'token' => $manageToken,
+            'url' => $kbcUrl,
+        ]);
+
+        $logger = new ConsoleLogger($output);
+        $queueApiUrl = str_replace('connection', 'queue', $kbcUrl);
+
+        $projects = $this->parseProjectIds($sourceFile);
+        $output->writeln(sprintf('Migrating "%s" projects', count($projects)));
+
+        $migrationJobs = [];
+        foreach ($projects as $projectId) {
+            // set queuev2 project feature
+            $projectRes = $manageClient->getProject($projectId);
```

Comment:
nějakej try catch když projekt neexistuje nebo je smazaný? aby to cely necrashlo

Reply:
jo,pridam

Reply:
![Screenshot 2021-09-24 at 13 06 02](https://user-images.githubusercontent.com/1488015/134664770-64e032c3-02d5-481f-99d2-6a46172bae58.png)